src/wp-includes/blocks/index.php

```diff
@@ -49,8 +49,34 @@ function register_core_block_style_handles() {
 		$files          = get_transient( $transient_name );
 		if ( ! $files ) {
 			$files = glob( wp_normalize_path( __DIR__ . '/**/**.css' ) );
+
+			/*
+			 * Make CSS files path relative before storing them in transient.
+			 *
+			 * Remove the `wp-includes` part to avoid issue when moving site (and the database) to another server :
+			 * /path/to/wp-includes/blocks/xxx/yyy.css => blocks/xxx/yyy.css
+			 */
+			$files = array_map(
+				static function ( $file ) use ( $includes_path ) {
+					return str_replace( $includes_path, '', $file );
+				},
+				$files
+			);
 			set_transient( $transient_name, $files );
 		}
```

Comment:
Regardless of whether we prepend `$includes_path` to all cached values here or not, we need to make sure to also consider the original (buggy) variant that may still be stored in cache. For example, check if the first element starts with the includes path already, and if so, strip it from all values, and then update the cache with the correct list (i.e. without the `$includes_path` prefix).

Alternatively, we'll have to implement an update routine for the release this will ship with, to `delete_transient( 'wp_core_block_css_files' )`.

Reply:
In https://github.com/WordPress/wordpress-develop/pull/5052/commits/f6cbed7bf8e72a61485f84da36af221a0909dd9c I've added a check which invalidate the transient is the stored values don't use the correct path.